### PR TITLE
enhancement: dynamic example and distinct role definition name

### PR DESCRIPTION
**Pull request by jaredfholgate**
This PR adds two enhancements:

1. It adds a post-fix to the role definition names to avoid clashes during testing or for real world scenarios.
2. It adds an end to end sample for using the module with a dynamically defined management group structure using a yaml file.

Additionally this PR brings this module in line with the latest AVM repo structure.

### CODE REVIEW

**Comment by jaredfholgate** [comment]
@matt-FFFFFF If you'd like me to get the E2E tests working (with a Ready to Test label setup), let me know.

File: examples/dynamic-input/header.md
Comment (matt-FFFFFF) [inline examples/dynamic-input/header.md:1]: This file is not required

File: examples/dynamic-input/footer.md
Comment (matt-FFFFFF) [inline examples/dynamic-input/footer.md:1]: This file is not required

**Review by matt-FFFFFF — CHANGES_REQUESTED** [review]
couple of files need removing, otherwise excellent!

**Comment by matt-FFFFFF** [comment]
> @matt-FFFFFF If you'd like me to get the E2E tests working (with a Ready to Test label setup), let me know.

That would be very useful I think feel free!

File: examples/dynamic-input/footer.md
Comment (jaredfholgate) [reply examples/dynamic-input/footer.md:1]: Should I remove these files from the other example too?

**Comment by jaredfholgate** [comment]
> > @matt-FFFFFF If you'd like me to get the E2E tests working (with a Ready to Test label setup), let me know.
> 
> That would be very useful I think feel free!

Tests are running now. There is a chance they will hit timing issue and will need to adjust the waits in the example. This is based on my experience earlier today.

**Comment by matt-FFFFFF** [comment]
awesome work @jaredfholgate LGTM

**Review by matt-FFFFFF — APPROVED** [review]

**Pull request merged**